models/article.models.js

```diff
@@ -51,3 +51,17 @@ exports.fetchArticles = async () => {
 
   return articles;
 };
+
+exports.fetchArticleCommentsById = async (id) => {
+  const { rows: comments } = await db.query(
+    "SELECT * FROM comments WHERE article_id = $1",
+    [id]
+  );
+  if (comments.length === 0) {
+    return Promise.reject({
+      status: 404,
+      msg: "No article found with that id!",
```

Comment:
`comments` being an empty array here doesn't guarantee that the article doesn't exist, it could just mean that there are no comments in a specific article_id. I recommend creating a separate utility function that checks if an id exists in a table and throws an error if it doesn't. You can then use this in a `Promise.all` in your controller.